### PR TITLE
Add NetBSD support.

### PR DESCRIPTION
**Pull request by 0-wiz-0**

### CODE REVIEW

File: ffi/build.py [hunk 1]
```diff
@@ -225,6 +225,8 @@ def main():
         main_posix('linux', '.so')
     elif sys.platform.startswith(('freebsd','openbsd')):
         main_posix('freebsd', '.so')
+    elif sys.platform.startswith('netbsd'):
+        main_posix('netbsd', '.so')
```
Comment (gmarkall) [inline ffi/build.py:229]: As you didn't add a `Makefile.netbsd`, how is this expected to work?

**Comment by 0-wiz-0** [comment]
I just forwarded an, it seems, cargo-culted patch. Sorry for the noise.

**Comment by gmarkall** [comment]
No worries - do you want this to work on netbsd? I'd guess changing line 226 of `build.py` (just above your changes) to

```python
elif sys.platform.startswith(('freebsd','openbsd', 'netbsd')):
```

has at least a chance of working. The other possibility is that in netbsd you could build with CMake by setting the environment variable `LLVMLITE_USE_CMAKE=1` to see if that "just works".

**Comment by 0-wiz-0** [comment]
Looking more closely, I found the second part of the change in the package:
```
sed -e 's/ -stdlib=libc++//' ${WRKSRC}/ffi/Makefile.freebsd > ${WRKSRC}/ffi/Makefile.netbsd
```
I.e. it uses Makefile.freebsd but removes `-stdlib=libc++`. I'm not sure why that was added though.

Earlier I tried a simple version bump update of the package but noticed a failure:
```
[100%] Running lit suite /scratch/devel/py-llvmlite/work/llvm/test/Unit
/scratch/devel/py-llvmlite/work/llvm-14.0.6.src/utils/lit/lit/formats/googletest.py:57: SyntaxWarning: invalid escape sequence '\('
  upstream_prefix = re.compile('Running main\(\) from .*gtest_main\.cc')
/scratch/devel/py-llvmlite/work/llvm-14.0.6.src/utils/lit/lit/TestRunner.py:181: SyntaxWarning: invalid escape sequence '\c'
  """
llvm-lit: /scratch/devel/py-llvmlite/work/llvm-14.0.6.src/utils/lit/lit/run.py:126: note: Raised process limit from 1024 to 1044
-- Testing: 6458 tests, 32 workers --
Testing:  0.. 10.. 20.. 30.. 40.. 50.. 60.. 70.
FAIL: LLVM-Unit :: Support/./SupportTests/FileSystemTest.RealPath (5034 of 6458)
******************** TEST 'LLVM-Unit :: Support/./SupportTests/FileSystemTest.RealPath' FAILED ********************
Script:
--
/scratch/devel/py-llvmlite/work/build/unittests/Support/./SupportTests --gtest_filter=FileSystemTest.RealPath
--
Note: Google Test filter = FileSystemTest.RealPath
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from FileSystemTest
[ RUN      ] FileSystemTest.RealPath
/scratch/devel/py-llvmlite/work/llvm/unittests/Support/Path.cpp:767: Failure
fs::real_path(HomeDir, Expected): did not return errc::success.
error number: 2
error message: No such file or directory

/scratch/devel/py-llvmlite/work/llvm/unittests/Support/Path.cpp:681: Failure
fs::remove(TestDirectory.str()): did not return errc::success.
error number: 66
error message: Directory not empty

[  FAILED  ] FileSystemTest.RealPath (0 ms)
[----------] 1 test from FileSystemTest (0 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (0 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] FileSystemTest.RealPath

 1 FAILED TEST
Test Directory: /tmp/lit-tmp-r1t29qpx/file-system-test-cdc719

********************
Testing:  0.. 10.. 20.. 30.. 40.. 50.. 60.. 70..
FAIL: LLVM-Unit :: Support/./SupportTests/FileSystemTest.permissions (5055 of 6458)
******************** TEST 'LLVM-Unit :: Support/./SupportTests/FileSystemTest.permissions' FAILED ********************
Script:
--
/scratch/devel/py-llvmlite/work/build/unittests/Support/./SupportTests --gtest_filter=FileSystemTest.permissions
--
Note: Google Test filter = FileSystemTest.permissions
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from FileSystemTest
[ RUN      ] FileSystemTest.permissions
/scratch/devel/py-llvmlite/work/llvm/unittests/Support/Path.cpp:2271: Failure
Expected equality of these values:
  fs::setPermissions(TempPath, fs::set_gid_on_exe)
    Which is: generic:1
  NoError
    Which is: system:0
/scratch/devel/py-llvmlite/work/llvm/unittests/Support/Path.cpp:2272: Failure
Value of: CheckPermissions(fs::set_gid_on_exe)
  Actual: false
Expected: true
/scratch/devel/py-llvmlite/work/llvm/unittests/Support/Path.cpp:2301: Failure
Expected equality of these values:
  fs::setPermissions(TempPath, fs::all_perms & ~fs::sticky_bit)
    Which is: generic:1
  NoError
    Which is: system:0
/scratch/devel/py-llvmlite/work/llvm/unittests/Support/Path.cpp:2303: Failure
Value of: CheckPermissions(fs::all_perms & ~fs::sticky_bit)
  Actual: false
Expected: true
[  FAILED  ] FileSystemTest.permissions (0 ms)
[----------] 1 test from FileSystemTest (0 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (0 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] FileSystemTest.permissions

 1 FAILED TEST
Test Directory: /tmp/lit-tmp-r1t29qpx/file-system-test-695917

********************
Testing:  0.. 10.. 20.. 30.. 40.. 50.. 60.. 70.. 80.. 90..
********************
Failed Tests (2):
  LLVM-Unit :: Support/./SupportTests/FileSystemTest.RealPath
  LLVM-Unit :: Support/./SupportTests/FileSystemTest.permissions


Testing Time: 6.90s
  Passed: 6456
  Failed:    2
--- test/CMakeFiles/check-llvm-unit ---
```
I also see this with 0.41.1.
(I've asked the previous packager for help.)

**Comment by gmarkall** [comment]
The test fails look like they're coming from a run of the LLVM test suite - do you know why that is building / running as part of the llvmlite build?

**Comment by 0-wiz-0** [comment]
Thanks. For some reason, the package was running the llvm test suite.
I've updated it to 0.43.0 now.
Should I make a pull request with the netbsd case, and a copy of the freebsd Makefile without libstdc++?

**Comment by gmarkall** [comment]
I'd welcome that - I can't guarantee it will stay working with NetBSD as we have no NetBSD CI, but at least it should save you patching it for now and could always submit any required fixes in the future.

**Pull request closed without merging**